### PR TITLE
Connect action with component Milestone 3 code review

### PR DESCRIPTION
**Pull request by dipbazz**
In this pull request, I have connected the redux action `createBook` and removeBook` to add and remove a book from the store.

Changes in the `BooksForm` component
-

- Made it stateful to have its own state with `title` and `category` of a book.
- Add methods 
    - `handleTitleChange` that handles the change in title input and store the value in the local state.
    - `handleCategoryChange` that handles the change in the category and store the value in the local state, and 
    - `handleSubmit` which submits the form and store the local title, and category to the redux store
- Add function `mapDispatchToProps` to map the `createBook` action to props which saves the local state to redux store.

changes in `BookList` and `Book` component
-

- Made `BookList` component to be stateful to take `removeBook` action as props
- Passed `handleRemoveBook` event to `Book` component as props
- `Book` component takes the `handleRemoveBook` function as props and dispatches the `removeBook` action on click event.

Any feedback and suggestions is appreciated.

### CODE REVIEW

File: src/containers/BooksForm.js [hunk 1]
```diff
@@ -1,28 +1,79 @@
...
+      title: e.target.value,
+    });
+  }
+
+  handleCategoryChange(e) {
+    this.setState({
+      category: e.target.value,
+    });
+  }
+
```
Comment (uche-inyama) [inline src/containers/BooksForm.js]: A single method like so, will be fine:

```
handleChange({ target: { name, value } }) {
  this.setState({
    [name]: value,
  })
}


```

You have to add **name attribute** to input and select tags in the form.

File: src/containers/BooksForm.js [hunk 2]
```diff
@@ -1,28 +1,79 @@
...
+            Book title
+            <input id="title" value={title} type="text" placeholder="Book title" onChange={this.handleTitleChange} required />
+          </label>
+        </div>
+
+        <div>
+          <label htmlFor="category">
+            Book category
+            <select id="category" onChange={this.handleCategoryChange} value={category} required>
+              {categories.map((item) => <option key={item} value={item}>{item}</option>)}
```
Comment (uche-inyama) [inline src/containers/BooksForm.js]: Remember to add **name attribute** to input and select tags.

File: src/containers/BooksForm.js [hunk 3]
```diff
@@ -1,28 +1,79 @@
...
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: 'Action',
+    };
+
+    this.handleTitleChange = this.handleTitleChange.bind(this);
+    this.handleCategoryChange = this.handleCategoryChange.bind(this);
```
Comment (uche-inyama) [inline src/containers/BooksForm.js]: Just a single bind would do: 

**``` this.handleChange = this.handleChange.bind(this)```**

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

### Hi dipbazz :wave: 

This milestone is coming along fine. But it would be nice if you refactor the handleChange method. I will leave an inline comment on how to go about it.

#### That's it, refactor those lines of code, then submit another form for a code review.

### Happy coding :man_technologist: 

#### ``` Uchechukwu Henry Inyama ```

File: src/containers/BooksForm.js [hunk 4]
```diff
@@ -1,28 +1,72 @@
...
+    this.state = {
+      title: '',
+      category: 'Action',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange({ target: { name, value } }) {
```
Comment (Forison) [inline src/containers/BooksForm.js]: **[Optional]**
React allows you to skip the binding step in line 14 and 15 if the functions are defined with the arrow function. This way you can have `handleChange=({ target: { name, value } })=>{ ` without binding in constructor. You may do same for `handleSubmit` as well

**Review by Forison — APPROVED** [review]
## Status::Approved✔️

#### Good job on refactoring the code as requested earlier, the code is much cleaner and more readable. Kindly go ahead and merge the branch and proceed to the next milestone👷.

#### Happy coding👨‍💻

Addo Forison

**Pull request merged**